quickjs.c

```diff
@@ -32272,6 +32273,24 @@ static int JS_WriteSharedArrayBuffer(BCWriterState *s, JSValueConst obj)
     return 0;
 }
 
+static int JS_WriteRegExp(BCWriterState *s, JSRegExp regexp)
+{
+    JSString *bc = regexp.bytecode;
+    assert(!bc->is_wide_char);
+
+    JS_WriteString(s, regexp.pattern);
+
+    if (is_be())
+        lre_byte_swap(bc->u.str8, bc->len, /*is_byte_swapped*/FALSE);
```

Comment:
I'm not sure I follow, why are we calling lre_byte_swap twice, first with false then true?

Reply:
is_byte_swapped tells lre_byte_swap whether the size field in the header is in host endianness or not, i.e., whether it needs to byte-swap it. It reads that in order to know how large the bytecode is, because bc->len == header + bytecode + group names.

Why twice? Do and undo: convert from BE to LE, write out data, convert back to BE.